pypeit/spectrographs/spectrograph.py

```diff
@@ -790,26 +793,36 @@ def get_meta_value(self, inp, meta_key, required=False, ignore_bad_header=False,
                 msgs.warn("Requested meta data for meta_key={} does not exist...".format(meta_key))
                 return None
 
+        # Is this meta required for this frame type (Spectrograph specific)
+        if ('required_ftypes' in self.meta[meta_key]) and (usr_row is not None):
+            required = False
+            for ftype in self.meta[meta_key]['required_ftypes']:
+                if ftype in usr_row['frametype']:
+                    required = True
+
         # Check if this meta key is required
         if 'required' in self.meta[meta_key].keys():
             required = self.meta[meta_key]['required']
 
         # Is this not derivable?  If so, use the default
         #   or search for it as a compound method
         value = None
-        if self.meta[meta_key]['card'] is None:
-            if 'default' in self.meta[meta_key].keys():
-                value = self.meta[meta_key]['default']
-            elif 'compound' in self.meta[meta_key].keys():
-                value = self.compound_meta(headarr, meta_key)
+        try:
```

Comment:
Try statements with 20 lines of code below them are both not good coding practice and very difficult to debug. I am as guilty of this as anyone, but my suggestion is that we try to purge as many try except blocks from this routine as possible, and if we do feel we have to use them, then only try: except: a single line of code (as best coding practice). I recently spent hours trying to find a trivial bug (going line by line) related to this routine that was obscured by  all the try: except: madness.

Reply:
This is not resolved. Please try and except the individual lines that can fail due to the following errors and not the entire code block.